### PR TITLE
We now have a pre update script

### PR DESCRIPTION
**Pull request by annando**
This script will run before the database structure check is executed.

This will be needed in PR https://github.com/friendica/friendica/pull/5383

### CODE REVIEW

File: boot.php [hunk 1]
```diff
@@ -731,25 +738,27 @@ function update_db()
 	return;
 }
 
-function run_update_function($x)
+function run_update_function($x, $post_update = true)
```
Comment (MrPetovan) [inline boot.php]: Please change the parameter from `bool $post_update` to `string $prefix = 'update'`. The prefix will be passed during the function call.

File: boot.php [hunk 2]
```diff
@@ -731,25 +738,27 @@ function update_db()
...
 {
-	if (function_exists('update_' . $x)) {
+	$funcname = ($post_update ? 'update_' : 'pre_update_');
```
Comment (MrPetovan) [inline boot.php]: Then `$funcname` should be `$funcname = $prefix . '_' . $x;`

File: boot.php [hunk 3]
```diff
@@ -731,25 +738,27 @@ function update_db()
...
+
+	if (function_exists($funcname . $x)) {
```
Comment (MrPetovan) [inline boot.php]: This `. $x` and all the subsequent ones can then be removed.

**Comment by annando** [comment]
@MrPetovan lucky? ;-)

**Comment by MrPetovan** [comment]
You know how to make me happy.

**Pull request merged**